analyzer/src/traversal/expressions.rs

```diff
@@ -641,15 +649,17 @@ fn expr_call_self_attribute(
     func_name: &str,
     args: &Node<Vec<Node<fe::CallArg>>>,
 ) -> Result<ExpressionAttributes, SemanticError> {
+    let called_func = scope
+        .borrow()
+        .contract_scope()
+        .borrow()
+        .function_def(func_name);
+
     if let Some(ContractFunctionDef {
         params,
         return_type,
         ..
-    }) = scope
-        .borrow()
-        .contract_scope()
-        .borrow()
-        .function_def(func_name)
+    }) = called_func
```

Comment:
Moving this out of the `if let` was needed to mitigate a `borrow_mut()` failure...this took me way to long to find :sweat_smile:

Reply:
yeah, Ive ran into this too